crates/api/src/func.rs

```diff
@@ -536,6 +601,16 @@ impl WasmTy for f32 {
     fn into_abi(self) -> Self::Abi {
         self
     }
+    #[inline]
+    unsafe fn load(ptr: &mut *const u128) -> Self::Abi {
+        let ret = f32::from_bits(*ptr as u32);
```

Comment:
Here and in the other `load` implementations, it looks like this is dereferencing the wrong thing.

Reply:
Oh dear good catch! Made sure to add tests to catch this